### PR TITLE
[Refactoring] move to using async function

### PR DESCRIPTION
**Pull request by oa-s**
sorry deleted branch by mistake

### CODE REVIEW

**Comment by oa-s** [comment]
added more functions to migration

**Comment by hboon** [comment]
> sorry deleted branch by mistake

Locally or in GitHub? If locally, you can try `git reflog` to find the commit to re-create it. If GitHub, you can restore it from the site

That didn't work?

File: AlphaWallet/ActiveWalletCoordinator.swift [hunk 1]
```diff
@@ -1250,10 +1251,10 @@ extension ActiveWalletCoordinator: DappBrowserCoordinatorDelegate {
                         session: WalletSession) -> AnyPublisher<String, PromiseError> {
 
         infoLog("[\(source)] ethCall")
-        return session.blockchainProvider
-            .call(from: from, to: to, value: value, data: data)
+        let provider = session.blockchainProvider
+        return Future { try await provider.call(from: from, to: to, value: value, data: data) }
             .receive(on: RunLoop.main)
-            .mapError { PromiseError(error: $0.unwrapped) }
+            .mapError { PromiseError(error: $0) }
```
Comment (hboon) [inline AlphaWallet/ActiveWalletCoordinator.swift:1257]: Does removing the `.unwrapped` change the behavior?

File: modules/AlphaWalletFoundation/AlphaWalletFoundation/RPC/BlockchainProvider.swift [hunk 2]
```diff
@@ -11,21 +11,24 @@ import AlphaWalletLogger
...
+    func balance(for address: AlphaWallet.Address) async throws -> Balance
+    func blockNumber() async throws -> Int
+    func transactionReceipt(hash: String) async throws -> TransactionReceipt
+    func call(from: AlphaWallet.Address?, to: AlphaWallet.Address?, value: String?, data: String) async throws -> String
     func call<R: ContractMethodCall>(_ method: R, block: BlockParameter) -> AnyPublisher<R.Response, SessionTaskError>
-    func pendingTransaction(hash: String) -> AnyPublisher<EthereumTransaction?, SessionTaskError>
-    func nextNonce(wallet: AlphaWallet.Address) -> AnyPublisher<Int, SessionTaskError>
-    func block(by blockNumber: BigUInt) -> AnyPublisher<Date, SessionTaskError>
+    func transaction(byHash hash: String) async throws -> EthereumTransaction?
+    func nonce(wallet: AlphaWallet.Address) async throws -> Int
```
Comment (hboon) [inline modules/AlphaWalletFoundation/AlphaWalletFoundation/RPC/BlockchainProvider.swift:26]: `getNextNonce(wallet:)` better?

**Review by hboon — APPROVED** [review]
Didn't check most of it.

**Pull request merged**